# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/sdn/sdnAccessible-inl.h
# Commit: a53fb045a8e3
# Full Hash: a53fb045a8e34c1e1f14ca7b705eb9a5a0e76db2
# Author: James Teh <jteh@mozilla.com>
# Date: 2021-05-02 09:29:52
# Regressor Bug: 1694865
# File Overlap Count: 1
# Description:
#   Bug 1694865 part 19: Make EnumVariant, GeckoCustom and sdn*Accessible classes aggregate MsaaAccessible/MsaaDocAccessible instead of AccessibleWrap/DocAccessibleWrap. r=morgan
#   
#   These objects need to aggregate the primary COM object (MsaaAccessible).
#   Once the IUnknown implementation moves out of AccessibleWrap, it won't be possible to aggregate *AccessibleWrap any more.
#   
# ==============================================================================

diff -r 5bd374b642fe -r a53fb045a8e3 accessible/windows/sdn/sdnAccessible-inl.h
--- a/accessible/windows/sdn/sdnAccessible-inl.h	Sat May 01 21:09:04 2021 +0000
+++ b/accessible/windows/sdn/sdnAccessible-inl.h	Sat May 01 22:29:25 2021 +0000
@@ -19,9 +19,9 @@
   return GetExistingDocAccessible(mNode->OwnerDoc());
 }
 
-inline AccessibleWrap* sdnAccessible::GetAccessible() {
-  if (mWrap) {
-    return mWrap;
+inline MsaaAccessible* sdnAccessible::GetMsaa() {
+  if (mMsaa) {
+    return mMsaa;
   }
 
   DocAccessible* document = GetDocument();
@@ -31,9 +31,10 @@
 
   // Once we have an accessible, we should hold a reference to it so that we
   // may preserve object identity.
-  mWrap = static_cast<AccessibleWrap*>(
+  AccessibleWrap* wrap = static_cast<AccessibleWrap*>(
       document->GetAccessibleEvenIfNotInMap(mNode));
-  return mWrap;
+  wrap->GetNativeInterface(getter_AddRefs(mMsaa));
+  return mMsaa;
 }
 
 }  // namespace a11y